### PR TITLE
colexec: remove duplication around bazel utility functions

### PR DESCRIPTION
**Pull request by yuzefovich**
Previously, we had essentially duplicated `.bzl` files in two packages
that need to generate code using `execgen`, but this can be avoided by
defining a separate `.bzl` file as an extension used in both places.

Release note: None

### CODE REVIEW

File: pkg/sql/colexecbase/EXECGEN.bzl [hunk 1]
```diff
@@ -71,3 +48,7 @@ def gen_eg_go_rules():
               """,
             tools = [":execgen", ":goimports"],
         )
+
+def rule_name_for(target, visibility=["//visibility:private"]):
```
Comment (irfansharif) [inline pkg/sql/colexecbase/EXECGEN.bzl:52]: Does this visibility annotation do anything? We're not really using it, no?

**Review by irfansharif — APPROVED** [review]

**Review by yuzefovich — COMMENTED** [review]
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/60993#-:-MUFIlbn4G4sWOC-7mfU:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @irfansharif and @rickystewart)

___
*[pkg/sql/colexecbase/EXECGEN.bzl, line 52 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/60993#-MUCiqZGAUZq0vcCC6Z9-r1-52:-MUFI_Gu-F0IXo1h66D8:b45tv72) ([raw file](https://github.com/cockroachdb/cockroach/blob/7d122db4c17c9828f0dfb849c542ee86ee648c12/pkg/sql/colexecbase/EXECGEN.bzl#L52)):*
<details><summary><i>Previously, irfansharif (irfan sharif) wrote…</i></summary><blockquote>

Does this visibility annotation do anything? We're not really using it, no?
</blockquote></details>

I thought it would prohibit from importing this function into another packages, but it seems like it doesn't do anything as you correctly pointed out, so I removed it.

**Review by rickystewart — APPROVED** [review]
wfm

**Comment by yuzefovich** [comment]
bors was saying there was a merge conflict, so I rebased and force-pushed (probably that was on a concurrent PR though).

bors r+

**Pull request merged**